substrate/client/authority-discovery/src/worker.rs

```diff
@@ -580,7 +580,7 @@ where
 
 				debug!(target: LOG_TARGET, "Value for hash '{:?}' found on Dht.", v.record.key);
 
-				if let Err(e) = self.handle_dht_value_found_event(v) {
+				if let Err(e) = self.handle_dht_value_found_event(v.into()) {
```

Comment:
You will also need to update `self.network.put_value(key, value)` to use network backend agnostic types all the way down to `NetworkWorker` (aka "libp2p network backend") / `Litep2pNetworkBackend`, where this type will be converted into a specific network backend type.

Reply:
Sorry, the `into` here was not needed. I made it so that `substrate/client/network/src/types.rs` does not use `libp2p` at all anymore. `DhtEvent` should also be free of any use of`libp2p`.
This Leaves us with `Litep2pNetworkBackend` which is still using `libp2p`. I will therefore start working on `substrate/client/network/src/litep2p/mod.rs` and all related files to remove `libp2p`.

Reply:
Sounds good! There should be no need for libp2p kademlia types in `substrate/client/network/src/litep2p/mod.rs` now when you have `kad.rs` with network backend agnostic types.

Ideally, we should get rid of all `libp2p` types in litep2p network backend, but better to do it as a follow-up PR.